h2o-genmodel/src/main/java/hex/genmodel/algos/ensemble/StackedEnsembleMojoReader.java

```diff
@@ -17,11 +22,64 @@ protected void readParentModelData() throws IOException {
         int baseModelNum = readkv("base_models_num", 0);
         _model._baseModelNum = baseModelNum;
         _model._metaLearner = getModel((String) readkv("metalearner"));
-        _model._baseModels = new MojoModel[baseModelNum];
+        _model._baseModels = new StackedEnsembleMojoModel.StackedEnsembleMojoSubModel[baseModelNum];
+        final String[] columnNames = readkv("[columns]");
+        ordinalModelHash = new HashMap<>();
+        int mappingHashOrdinal = 0;
         for (int i = 0; i < baseModelNum; i++) {
             String modelKey = readkv("base_model" + i);
-            _model._baseModels[i] = getModel(modelKey);
+            final MojoModel model = getModel(modelKey);
+            final int[] mapping = createMapping(model, columnNames, modelKey);
+            final int modelMappingIndex;
+            final int modelMappingHashcode = Arrays.hashCode(mapping);
```

Comment:
Why do we need to work with hash code? Why can the model not be identified by the index of the model? The order of the models is stable after the reader finishes.

Reply:
okay, I see what you are doing after reading the PR

you can still hit a hash collision, to be 100% correct you would need to prevent them